### ISSUE TITLE
name() function is not implemented in DefaultKeyType.

### ISSUE THREAD

**Issue by snanurag** [google/keyczar#170]
I am trying to create my own EncryptReader and getting following error.

Exception in thread "main" java.lang.Error: Unresolved compilation problem: 
    The method getName() of type DefaultKeyType must override a superclass method

```
at org.keyczar.DefaultKeyType.getName(DefaultKeyType.java:140)
at org.keyczar.KeyMetadata.toString(KeyMetadata.java:85)
at EgainEncryptReader.getMetadata(test.java:79)
at org.keyczar.Keyczar.<init>(Keyczar.java:79)
at org.keyczar.Encrypter.<init>(Encrypter.java:58)
at test.main(test.java:35)
```

I am using DefaultKeyType.AES in my EncryptReader constructor. I believe, this is not a huge error. Can someone commit its fix asap? I really need it urgently. 

Thanks,
Anurag

**Comment by sweis** [comment]
Are you using Java 1.5? I just cloned the latest repo as a Maven project in Eclipse and it was complaining about the "@Override" annotations. In Java 1.5, I think you can't use the @Override on methods implementing interfaces.

Try building with a later version of Java. That fixed the problem for me.

The Maven pom.xml file seems to specify Java 1.5, which probably caused Eclipse to configure the project to use that version.

**Comment by sweis** [comment]
Making this change and reimporting the project into Maven addressed the issue for me:
$ git diff
diff --git a/java/code/pom.xml b/java/code/pom.xml
index 700c14e..7128ae1 100644
--- a/java/code/pom.xml
+++ b/java/code/pom.xml
@@ -121,8 +121,8 @@
         <artifactId>maven-compiler-plugin</artifactId>
        <version>2.3.2</version>
         <configuration>
-          <source>1.5</source>
-          <target>1.5</target>
-          <source>1.6</source>
-          <target>1.6</target>
       </configuration>
     </plugin>

**Comment by snanurag** [comment]
I think, I did not explain it well. 

What I have done exactly is, I have overwritten getMetadata() function in my implementation of EncryptedReader. In that, I have implemented getMetadata() in the following way.

	public String getMetadata() throws KeyczarException
	{
			KeyMetadata metaData = new KeyMetadata("eGain", KeyPurpose.DECRYPT_AND_ENCRYPT, DefaultKeyType.AES);
			
			return metaData.toString();
	}

The error is coming because, java.lang.Enum class has abstract method name() method which is called in getName() method of DefaultKeyType. Either abstract method name should be implemented or getName() method of DefaultKeyType should have some other implementation e.g. toString().

**Comment by snanurag** [comment]
Hi Sweis,

Thanks for looking into the issue. It worked for me as well. 

But, I am facing further problems. I don't want to keep my key open in the file system. I want to keep it password protected in a Keystore (java.security.KeyStore). Then, I want to read it in memory. I want to get my key from Keystore in memory only. I will be keeping my Keystore and password on file system. For this, I was trying to implement EncryptedReader functions. I used AesKey and KeyMetadata class constructors to build my key and meta data file. These, I used in implementing functions of EncryptedReader. But I am still facing errors. Right now, I am at this error :

Exception in thread "main" org.keyczar.exceptions.NoPrimaryKeyException: No primary key found
    at org.keyczar.Encrypter.ciphertextSize(Encrypter.java:89)
    at org.keyczar.Encrypter.encrypt(Encrypter.java:110)
    at org.keyczar.Encrypter.encrypt(Encrypter.java:182)
    at test.main(test.java:31)

I have seen almost all the classes in this API, but it doesn't sound like that I can work without keeping key in file system. Is there anyway that I could save myself from keeping key in file system? 

PS: I am looking for AES encryption.
